src/main/java/uk/gov/companieshouse/controller/PaymentController.java

```diff
@@ -48,12 +48,12 @@ public PaymentController(
     })
     @GetMapping()
     @ResponseStatus(HttpStatus.OK)
-    public PaymentGetResponse getPaymentUIData(@PathVariable("company-number") final String companyNumber) {
+    public PaymentGetResponse getPaymentUIData(@PathVariable("application-reference") final String applicationReference) {
         DissolutionGetResponse dissolutionInfo = dissolutionService
-                .getByCompanyNumber(companyNumber)
+                .getByApplicationReference(applicationReference)
                 .orElseThrow(NotFoundException::new);
 
-        return paymentService.get(dissolutionInfo.getETag(), dissolutionInfo.getApplicationType(), companyNumber);
+        return paymentService.get(dissolutionInfo, applicationReference);
```

Comment:
can the application reference be extracted from the dissolutionInfo rather than being passed in separately?